### PR TITLE
Recommend smbios

### PR DESCRIPTION
**Pull request by vstehle**
Start easy and just add details about the SMBIOS tables and structures, with a reference to the SMBIOS specification v3.7.0.
Should we remove the conditional and require SMBIOS?

### CODE REVIEW

**Comment by nullr0ute** [comment]
> Should we remove the conditional and require SMBIOS?

Yes, I think we should. A lot of support tools use SMBIOS to provide information to users in a GUI/TUI and collect information for support cases. By propagating that information the existing versions of these tools will just function as expected and we don't have to try and boil the ocean to get everything updated to support other mechanisms. It's also a requirement of ServerReady so it allows alignment across all SystemReady certs.

**Comment by nullr0ute** [comment]
Example tool is [sosreport](https://github.com/sosreport/sos) and an example issue with a [GUI management tool](https://github.com/cockpit-project/cockpit/issues/16972)

**Comment by vstehle** [comment]
Rebased.

**Comment by nullr0ute** [comment]
Should we have somewhere in the spec around suggesting/requiring SMBIOS tables to be presented and complete/accurate?

**Comment by nullr0ute** [comment]
Also this is useful:
https://developer.arm.com/documentation/102677/0100/SMBIOS-requirements

**Comment by samerhaj** [comment]
One solution is for EBBR to reference specific items from [Base Boot Requirements (BBR) specification](https://developer.arm.com/documentation/den0044/), section 9 - SMBIOS REQUIREMENTS.

It is also possible to do it the other way around (or both ways), where the BBR specification (which already references EBBR spec, when defining the EBBR receipe) can detail the SMBIOS requirements for such systems. I can help with adding that language (or cross-reference) in BBR when needed.

**Comment by vstehle** [comment]
Rebased.

**Comment by vstehle** [comment]
Moving to draft after [call of 11 Mar](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2024.03.11).

**Comment by vstehle** [comment]
Respun:
- Only "strongly recommend" SMBIOS, and when implemented require structures
- Remove Type 32 from table of required structures
- Warn that SMBIOS will be required in the future
- Add a note to clarify that firmware can provide more than what is required

**Comment by vstehle** [comment]
Trim down even more: only recommend SMBIOS and do not require any specific structure or table.

**Review by xypron — CHANGES_REQUESTED** [review]

File: source/chapter2-uefi.rst [hunk 1]
```diff
@@ -385,6 +385,26 @@ The DTB must be contained in memory of type `EfiACPIReclaimMemory`.
...
+------
+
+It is strongly recommended that firmware provides SMBIOS tables and structures.
+[#SMBNote]_
+
+If firmware provides SMBIOS tables and structures then it must provide them as
+described in :UEFI:`4.6.1.1` and [SMBIOS]_ § 5 & 6.
+
+.. warning:: A future version of this specification will require firmware to
+   provide SMBIOS tables and strutures and will explicitly list which ones are
```
Comment (xypron) [inline source/chapter2-uefi.rst:398]: %s/strutures/structures/

**Comment by vstehle** [comment]
Dropping this pull request after [the call of June 17](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2024.06.17); there is not enough remaining in the pull to be interesting.
Let's revive it later on.

**Pull request closed without merging**